src/main/java/org/spongepowered/api/item/inventory/Slot.java

```diff
@@ -35,4 +37,18 @@
      * @return the viewed inventory slot or itself if not a container slot
      */
     Slot viewedSlot();
+
+    /**
+     * Adds the ItemStack to this slot overwriting the existing item.
```

Comment:
Per your docs, this is really a consume method and not just a set if your ItemStack object is not being used by the Slot.